src/layer_norm.cpp

```diff
@@ -0,0 +1,333 @@
+{
+    if(dx == nullptr || x == nullptr || mean == nullptr || rstd == nullptr || dy == nullptr)
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "Null pointer for tensor.");
+    }
+
+    if(dxDesc.GetType() != dyDesc.GetType())
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "Tensor types do not match.");
+    }
+
+    if(dxDesc.GetLengths() != dyDesc.GetLengths())
+    {
+        MIOPEN_THROW(miopenStatusBadParm, "Tensor dimension lengths do not match.");
+    }
+
+    if(miopen::CheckNumericsEnabled())
+    {
+        miopen::checkNumericsInput(handle, xDesc, x);
+    }
+
+    bool is_all_packed = xDesc.IsPacked() && dyDesc.IsPacked() && weightDesc.IsPacked() &&
+                         meanDesc.IsPacked() && rstdDesc.IsPacked() && dxDesc.IsPacked() &&
+                         dwDesc.IsPacked() && dbDesc.IsPacked();
+
+    auto dims         = xDesc.GetLengths();
+    size_t grid_size  = 1;
+    size_t outer_size = 1;
+    size_t inner_size = 1;
+    size_t i          = 0;
+    for(; i < normalized_dim; i++)
+    {
+        outer_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    for(; i < dims.size(); i++)
+    {
+        inner_size *= dims[i];
+        grid_size *= dims[i];
+    }
+
+    auto dtype = xDesc.GetType();
+
+    const std::vector<size_t> vld{LOCAL_SIZE, 1, 1};
+    const std::vector<size_t> vgd{outer_size * vld[0], 1, 1};
+
+    std::string algo_name = "LayerNormBackward";
+    std::string network_config =
+        "lnbwd-dtype" + std::to_string(static_cast<int>(dtype)) + "g" + std::to_string(vgd[0]) +
+        "l" + std::to_string(vld[0]) + "normalized_dim" + std::to_string(normalized_dim) + "grid" +
+        std::to_string(grid_size) + "outer_size" + std::to_string(outer_size) + "inner_size" +
+        std::to_string(inner_size) + "xpk" + std::to_string(static_cast<int>(xDesc.IsPacked())) +
+        "dypk" + std::to_string(static_cast<int>(dyDesc.IsPacked()));
+    if(weight)
+        network_config += "weightpk" + std::to_string(static_cast<int>(weightDesc.IsPacked()));
+    if(mean)
+        network_config += "meanpk" + std::to_string(static_cast<int>(meanDesc.IsPacked()));
+    if(rstd)
+        network_config += "rstdpk" + std::to_string(static_cast<int>(rstdDesc.IsPacked()));
+    network_config += "dxpk" + std::to_string(static_cast<int>(dxDesc.IsPacked())) + "mode" +
+                      std::to_string(static_cast<int>(mode));
+
+    std::string program_name = "MIOpenLayerNorm.cpp";
+    std::string kernel_name  = "LayerNormBwdContiguous";
+
+    // compile parameters
+    std::string parms =
+        " -DMIOPEN_USE_FP16=" + std::to_string(static_cast<int>(dtype == miopenHalf)) +
+        " -DMIOPEN_USE_FP32=" + std::to_string(static_cast<int>(dtype == miopenFloat)) +
+        " -DMIOPEN_USE_FP64=" + std::to_string(static_cast<int>(dtype == miopenDouble)) +
+        " -DMIOPEN_USE_BF16=" + std::to_string(static_cast<int>(dtype == miopenBFloat16));
```

Comment:
I would suggest to pass `-DMIOPEN_TYPE=` directly without those intermediate defines.